code/modules/events/apc_short.dm

```diff
@@ -27,51 +27,105 @@
+					continue
+				// if we are going to break this one
+				if(prob(APC_BREAK_PROBABILITY))
+					// if it has internal wires, cut the power wires
+					if(A.wires)
+						if(!A.wires.is_cut(WIRE_MAIN_POWER1))
+							A.wires.cut(WIRE_MAIN_POWER1)
+						if(!A.wires.is_cut(WIRE_MAIN_POWER2))
+							A.wires.cut(WIRE_MAIN_POWER2)
+					// if it was operating, toggle off the breaker
+					if(A.operating)
+						A.toggle_breaker()
+					// no matter what, ensure the area knows something happened to the power
+					current_area.power_change()
+					affected_apc_count++
+			log_and_message_admins("APC Short event shorted out [affected_apc_count] APCs.")
 
 /proc/power_restore(announce=TRUE)
-	if(announce)
-		GLOB.event_announcement.Announce("Power has been restored to [station_name()]. We apologize for the inconvenience.", "Power Systems Nominal", new_sound = 'sound/AI/poweron.ogg')
-
-	// recharge the APCs
-	for(var/thing in GLOB.apcs)
-		var/obj/machinery/power/apc/A = thing
-		if(!is_station_level(A.z))
-			continue
-		var/obj/item/stock_parts/cell/C = A.get_cell()
-		if(C)
-			C.give(C.maxcharge)
+	switch(alert("What Would You Like to Do?", "Make All Areas Powered", "Power all APCs", "Repair all APCs", "Repair and Power APCs"))
+		if("Power all APCs")	//Power without Repairing
+			if(announce)
+				GLOB.event_announcement.Announce("All operational APCs on [station_name()] have been fully charged.", "Power Systems Nominal", new_sound = 'sound/AI/poweron.ogg')
+			var/affected_apc_count = 0
+			for(var/obj/machinery/power/apc/A in GLOB.apcs)
+				var/obj/item/stock_parts/cell/C = A.get_cell()
+				var/area/current_area = get_area(A)
+				if(!is_station_level(A.z))
+					continue
+				if((length(A.wires.cut_wires) == 0) && A.operating && !A.shorted)
+					if(C)
+						C.charge = C.maxcharge
+				affected_apc_count++
+				current_area.power_change()
+			log_and_message_admins("Power has been restored to [affected_apc_count] APCs.")
+		if("Repair all APCs")	//Repair without charging
+			if(announce)
+				GLOB.event_announcement.Announce("All APCs on [station_name()] have been repaired.", "Power Systems Nominal", new_sound = 'sound/AI/poweron.ogg')
+			for(var/obj/machinery/power/apc/A in GLOB.apcs)
+				var/area/current_area = get_area(A)
+				if(!is_station_level(A.z))
+					continue
+				if(A.wires)
+					A.wires.repair()
+				if(!A.operating)
+					A.toggle_breaker()
+				if(A.shorted)
+					A.shorted = FALSE
+				current_area.power_change()
+			log_and_message_admins("Power has been restored to all APCs.")
+		if("Repair and Power APCs")	//Repair and Power APCs
+			if(announce)
+				GLOB.event_announcement.Announce("All APCs on [station_name()] have been repaired and recharged. We apologize for the inconvenience.", "Power Systems Optimal", new_sound = 'sound/AI/poweron.ogg')
+			// repair the APCs and recharge them
+			for(var/obj/machinery/power/apc/A in GLOB.apcs)
+				var/obj/item/stock_parts/cell/C = A.get_cell()
```

Comment:
Same thing here, declare this when you need it, sometime after the `continue` below.